design/design-doc-template-stacks-experience.md

```diff
@@ -0,0 +1,473 @@
+  desired.
+* A **template** is a chunk of configuration which will be reused by
+  the Template Stack to create configuration.
+* A **render** is the configuration which is the output of the stack's
+  controller in response to a render request. The configuration is
+  created by processing the stack's input templates using the desired
+  engine, and using the render request for context.
+* **Behaviors** are the configuration settings for a Template Stack.
+  For example, so that its controller knows how to respond to a render
+  request of a known type.
+
+## How to read this document
+
+This document is intended to be read alongside the [template stacks
+experience repository](https://github.com/suskin/template-stack-experience),
+which contains realistic code samples for different pieces of the
+functionality.
+
+Additionally, for more detail about the internals of template stacks on
+the stack manager and Crossplane side, see the [design doc focused on
+this](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-template-stacks.md)
+
+## Design
+
+The overall flow of authoring and consuming a template stack will be
+very similar for all scenarios. It is expected that template stacks will
+support a wide variety of scenarios, so this section will show some
+examples to help explain them.
+
+For a more complete treatment of example scenarios, see the [template
+stack experience repository](https://github.com/suskin/template-stack-experience).
+
+### Writing
+
+Creating a template stack from scratch involves only a couple steps. The
+project must first be initialized, and then the yamls must be put in a
+particular directory. The rbac requirements for the stack must also be
+configured, though some of that is done by the stack manager at runtime.
+At a high level, the steps would be as follows:
+
+1. Create a project directory.
+2. In the project directory, `kubectl crossplane stack init --template
+   myorg/mystack` to create the boilerplate of the stack layout.
+3. Relative to the project directory, put the yamls in
+   `config/stack/manifests/resources/templates` .
+4. Add CRD definitions; this could be done by putting CRD yaml
+   definitions in `config/stack/manifests/resources` or in
+   `config/crd/bases`.
+5. Add configuration to `stack.yaml` to
+   specify how configurations are rendered.
+6. Edit `config/stack/manifests/resources/app.yaml` to specify that the
+   stack will be working with all of the kinds that it will be working
+   with (using the `dependsOn` field).
+7. Edit `config/stack/manifests/resources/app.yaml` as appropriate.
+
+#### Adding a CRD
+
+The tooling will make it simpler to add a CRD from scratch. For example,
+the following would create a basic CRD in the appropriate folder, and
+would update the `dependsOn` field of the stack's `app.yaml`:
+
+```
+kubectl crossplane stack crd init WordpressInstance wordpressinstances wordpress.samples.stacks.crossplane.io
```

Comment:
I believe this would create an `owns` relationship.  A dependsOn relationship does not result in the inclusion of a new physical CRD file in the Stack, merely a line item in the app.yaml (which results in RBAC rules permitting the Stack access to the resources it depends on).

Reply:
This is changed